apps/backend/src/authn/okta.strategy.ts

```diff
@@ -26,50 +25,54 @@ export class OktaStrategy extends PassportStrategy(Strategy, 'okta') {
   ) {
     super(
       {
-        issuer: `https://${
-          configService.get('OKTA_DOMAIN') || 'disabled'
-        }/oauth2/default`,
-        authorizationURL: `https://${
-          configService.get('OKTA_DOMAIN') || 'disabled'
-        }/oauth2/default/v1/authorize`,
-        tokenURL: `https://${
-          configService.get('OKTA_DOMAIN') || 'disabled'
-        }/oauth2/default/v1/token`,
-        userInfoURL: `https://${
-          configService.get('OKTA_DOMAIN') || 'disabled'
-        }/oauth2/default/v1/userinfo`,
+        issuer:
+          configService.get('OKTA_ISSUER_URL') ||
+          `https://${configService.get('OKTA_DOMAIN')}` ||
+          'disabled',
+        authorizationURL:
+          configService.get('OKTA_AUTHORIZATION_URL') ||
+          `https://${configService.get('OKTA_DOMAIN') || 'disabled'}/oauth2/v1/authorize`,
+        tokenURL:
+          configService.get('OKTA_TOKEN_URL') ||
+          `https://${configService.get('OKTA_DOMAIN') || 'disabled'}/oauth2/v1/token`,
+        userInfoURL:
+          configService.get('OKTA_USER_INFO_URL') ||
+          `https://${configService.get('OKTA_DOMAIN') || 'disabled'}/oauth2/v1/userinfo`,
         clientID: configService.get('OKTA_CLIENTID') || 'disabled',
         clientSecret: configService.get('OKTA_CLIENTSECRET') || 'disabled',
-        callbackURL: `${configService.get('EXTERNAL_URL')}/authn/okta/callback`,
-        scope: 'openid email profile'
+        callbackURL: `${configService.get('EXTERNAL_URL')}/authn/okta_callback`,
```

Comment:
Was switching the `okta/callback` to `/okta_callback` stylistic, or did it change library use or functionality or something?

Reply:
It seems like Okta might have some weird bug in the resolution of the callback path where it tries to go to /okta instead of /okta/callback for some reason iirc.  Workaround is just to flatten the path.

See: https://github.com/okta/okta-oidc-middleware/issues/20

Reply:
So from our own docs on this, we say `Your callback URL should be https://<your.heimdall.instance>/authn/okta/callback`. Does that mean, for example, a new release of this code would require us to update existing okta integrations accordingly?

Reply:
Yes, this will be a breaking change for our OKTA and OIDC users.  We will have to update the minor version number on our next release and state that this change is coming.

Reply:
Okay, very important to know.